### ISSUE TITLE
Profile provider doesn't work with `aws configure sso` output

### ISSUE THREAD

**Issue by GTB3NW** [awslabs/aws-sdk-rust#741]
### Describe the bug

The file provider does not work with credentials configured with `aws configure sso`. Instead you must copy the following fields into the profile (rather than sourcing the session)

- `sso_region`
- `sso_start_url`
- `sso_registration_scopes`

and remove the source of the sso-session. At which point you then have to `aws sso login` to get some fresh credentials in the cache.

### Expected Behavior

Considering this is an official project, the file provider should work with the standardised profile and sso-session configuration

### Current Behavior

It does not inherit the settings from the source

### Reproduction Steps

- `aws configure sso`
- **try use the profile with simple provider snippet**
^ Bug should be pro

### Possible Solution

- copy `sso_region` into profile
- copy `sso_start_url` into profile
- copy `sso_registration_scopes` into profile
- remove `source`
- optionally remove the sso-session
- `aws sso login`

manual fix of aws config required to avoid the issue for now, but this should work with standard setups that the aws cli generates

### Additional Information/Context

I created my profile as default, but other profile names do not work either. My suspicion is it's the `source` of the sso-session which it's not merging the required sso fields in

### Version

```text
aws-config = "0.54.1"
aws-sdk-ec2 = "0.24.0"
aws-types= "0.54.1"
```

### Environment details (OS name and version, etc.)

aws-cli/2.9.15

### Logs

Error: Unhandled(Unhandled { source: ConstructionFailure(ConstructionFailure { source: CredentialsStageError { source: InvalidConfiguration(InvalidConfiguration { source: "ProfileFile provider could not be built: profile `default` was not defined: `sso_region` was missing" }) } }) })

**Comment by Velfi** [comment]
Hey @GTB3NW, thanks for submitting this issue. We'll add it to our backlog.

**Comment by ysaito1001** [comment]
Hello, thank you for reporting this! Support for `sso-session` has not been implemented but we keep track of it via https://github.com/awslabs/aws-sdk-rust/issues/703. Meanwhile, feel free to submit a PR to add support.

**Comment by GTB3NW** [comment]
> Hello, thank you for reporting this! Support for `sso-session` has not been implemented but we keep track of it via #703. Meanwhile, feel free to submit a PR to add support

Perfect thank you. I'm subscribed to that thread and look forward to it. I'll close this.
